lib/otel/span-processor.js

```diff
@@ -73,8 +82,61 @@ module.exports = class NrSpanProcessor {
       this.reconcileDbAttributes({ segment, span })
     } else if (span.kind === SpanKind.PRODUCER) {
       this.reconcileProducerAttributes({ segment, span })
+    } else if (span.kind === SpanKind.CLIENT && (span.attributes[ATTR_HTTP_METHOD] || span.attributes[ATTR_HTTP_REQUEST_METHOD])) {
+      this.reconcileHttpExternalAttributes({ segment, span })
+    }
+  }
+
+  reconcileHttpExternalAttributes({ segment, span }) {
+    for (const [prop, value] of Object.entries(span.attributes)) {
+      let key = prop
+      let sanitized = value
+
+      if (key === ATTR_HTTP_REQUEST_METHOD || key === ATTR_HTTP_METHOD) {
+        key = 'procedure'
+      } else if (key === ATTR_SERVER_ADDRESS || key === ATTR_NET_PEER_NAME) {
+        key = 'hostname'
+        sanitized = this.sanitizeHostname(sanitized)
+        segment.addSpanAttribute(key, sanitized)
+      } else if (key === ATTR_HTTP_HOST) {
+        key = 'host'
+        sanitized = this.sanitizeHostname(sanitized)
+      } else if (key === ATTR_SERVER_PORT || key === ATTR_NET_PEER_PORT) {
+        key = 'port'
+        segment.addSpanAttribute(key, sanitized)
+      } else if (key === ATTR_HTTP_RESP_STATUS_CODE || key === ATTR_HTTP_STATUS_CODE) {
+        key = 'http.statusCode'
+        segment.addSpanAttribute(key, sanitized)
+      } else if (key === ATTR_HTTP_STATUS_TEXT) {
+        key = 'http.statusText'
+        segment.addSpanAttribute(key, sanitized)
+      } else if (key === ATTR_URL_SCHEME || key === ATTR_HTTP_URL) {
+        sanitized = this.constructUrl(span)
+        key = 'url'
+      } else if (key === ATTR_URL_QUERY || key === ATTR_HTTP_TARGET) {
+        this.addQueryParameters(segment, sanitized)
+      }
+      segment.addAttribute(key, sanitized)
+    }
+  }
+
+  sanitizeHostname(hostname) {
+    return urltils.isLocalhost(hostname) ? this.agent.config.getHostnameSafe(hostname) : hostname
+  }
+
+  constructUrl(span) {
```

Comment:
you don't have to construct the URL. you can either use `url.full` or `http.url`, parse, scrub and obfuscate without qp's

Reply:
Ok. But I still need to get the host since `scrubAndParseParameters` doesn't get the host. It only gets the protocol, path and parameters.